CONTRIBUTING.md

```diff
@@ -60,6 +60,91 @@ bazel build //examples/simple:example_simple
 bazel-bin/examples/simple/example_simple
 ```
 
+### DevContainer Setup for Project
+
+This guide provides instructions on how to set up and use the development
+container (`devcontainer`) environment to streamline testing and development
+for this project. With the DevContainer, you can work in a consistent environment
+configured with all the necessary dependencies and tools.
+
+#### Prerequisites
+
+Before getting started, ensure you have the following installed:
+
+* **Docker**: DevContainers require Docker for containerization.
+* **Visual Studio Code (VSCode)** with the **Remote - Containers** extension.
```

Comment:
Haven't used DevContainers before, but I imagine it would be possible to do this without relying on VS Code ?

Please correct me if I'm wrong, but I imagine we could simply exec into a running docker container as well and run CI tests from there ?

I would prefer if we not put in instructions that are IDE specific.